pretty_cron/api.py

```diff
@@ -157,7 +168,46 @@ def _ordinal(num):
     return _human_list(ordinal_days)
 
 
+def _pretty_time_with_steps(minute, hour):
+    if minute['stepexpr']:
+        start = int(minute['start'])
+        end = int(minute['end'] or 59)
+        step = _ordinal(int(minute['step']))
+
+        hour_expr = ""
+        if hour['allexpr'] != '*':
+            hour_expr = " past hour {0}".format(hour['valexpr'])
+        return "Every {step} minute from {m_start} through {m_end}{hour_expr}".format(
+            step=step,
+            m_start=start,
+            m_end=end,
+            hour_expr=hour_expr)
+
+    start = int(hour['start'])
+    end = int(hour['end'] or 23)
+    step = _ordinal(int(hour['step']))
+
+    if minute['allexpr']:
+        min_expr = "Every minute"
+    else:
+        minval = int(minute['valexpr'])
+        min_expr = "Every {0} minute".format(_ordinal(minval)) if minval != 0 else "At the start"
+    return "{min_expr} of every {step} hour from {m_start} through {m_end}".format(
+        step=step,
+        m_start=start,
+        m_end=end,
+        min_expr=min_expr)
+
+
 def _pretty_time(minute, hour):
+    min_groups, hour_groups = (m and m.groupdict() for m in (cron_pattern.match(str(p)) for p in (minute, hour)))
+    # see if either the min or hour are a step expression
+    step_count = sum(bool(g['stepexpr']) for g in (min_groups, hour_groups))
+    if step_count > 0:
+        if step_count == 2:
+            raise ValueError('Step expressions only work for either minutes or hours, but not both at this time.')
```

Comment:
The API of the function is, as per the docs, to return a formatted string, or the original cron string if there's a feature that's not supported. `ValueError` should not be raised.